moped-editor/src/views/dashboard/DashboardView.js

```diff
@@ -125,170 +241,89 @@ const DashboardView = () => {
+    if (TABS[activeTab].label === "Following" && !!data) {
+      selectedData = data.moped_user_followed_projects;
+    } else if (TABS[activeTab].label === "My projects" && !!data) {
+      selectedData = data.moped_proj_personnel;
+    }
+
+    if (selectedData) {
       /**
-       * Get percentage of milestones completed
+       * Build data needed in Dashboard Material Table
        */
-      const milestonesTotal = project.project.moped_proj_milestones.length;
-      const milestonesCompleted = project.project.moped_proj_milestones.filter(
-        (milestone) => milestone.completed === true
-      ).length;
-      project["completed_milestones_percentage"] = !!milestonesTotal
-        ? (milestonesCompleted / milestonesTotal) * 100
-        : 0;
+      selectedData.forEach((project) => {
+        project["project_name_full"] = project.project.project_name_full;
+        project["project_id"] = project.project.project_id;
+        project["phase_name"] =
+          project.project.moped_proj_phases?.[0]?.moped_phase.phase_name;
+        project["phase_key"] =
+          project.project.moped_proj_phases?.[0]?.moped_phase.phase_key;
+        project["current_phase_id"] =
+          project.project.moped_proj_phases?.[0]?.moped_phase.phase_id;
+        /**
+         * Get percentage of milestones completed
+         */
+        const milestonesTotal = project.project.moped_proj_milestones.length;
+        const milestonesCompleted =
+          project.project.moped_proj_milestones.filter(
+            (milestone) => milestone.completed === true
+          ).length;
+        project["completed_milestones_percentage"] = !!milestonesTotal
+          ? (milestonesCompleted / milestonesTotal) * 100
+          : 0;
 
-      // project status update equivalent to most recent project note
-      // html is parsed before being rendered in the DashboardStatusModal component
-      project["status_update"] = "";
-      if (project?.project?.moped_proj_notes?.length) {
-        const note = project.project.moped_proj_notes[0]["project_note"];
-        project["status_update"] = note ? note : "";
-      }
-    });
-  }
+        project["id"] = project.project.project_id;
 
-  /** Build custom user greeting
-   */
-  const date = new Date();
-  const curHr = format(date, "HH");
-  const dateFormatted = format(date, "EEEE - LLLL dd, yyyy");
-  const getTimeOfDay = (curHr) => {
-    switch (true) {
-      case curHr < 12:
-        return "morning";
-      case curHr >= 12 && curHr < 18:
-        return "afternoon";
-      default:
-        return "evening";
+        // project status update equivalent to most recent project note
+        // html is parsed before being rendered in the DashboardStatusModal component
+        project["status_update"] = "";
+        if (project?.project?.moped_proj_notes?.length) {
+          const note = project.project.moped_proj_notes[0]["project_note"];
+          project["status_update"] = note ? note : "";
```

Comment:
i'm struggling a bit with this block. is note ever null/undefined, and why is an empty string preferred?